geomesa-core/src/main/scala/org/locationtech/geomesa/core/csv/package.scala

```diff
@@ -167,14 +156,14 @@ package object csv extends Logging {
 
     val latlonIdx = latlonFields.map { case (latf, lonf) => (idxOfField(latf), idxOfField(lonf)) }
     val fb = new SimpleFeatureBuilder(sft)
-    val parsers = sft.getTypes.map { t => fieldParserMap(t.getBinding) }
+    val parsers = sft.getTypes.map(t => getParser(t.getBinding))
     val fc = new DefaultFeatureCollection
     val records = CSVFormat.DEFAULT.parse(reader).iterator()
     if (hasHeader) { records.next } // burn off header rather than try (and fail) to parse it.
-    for {
-      record <- records
-      f      <- buildFeature(record, fb, parsers, latlonIdx) // logs and discards lines that fail to parse but keeps processing
-    } fc.add(f)
+    records.foreach { record =>
+      // logs and discards lines that fail to parse but keeps processing
+      buildFeature(record, fb, parsers, latlonIdx).foreach(fc.add)
+    }
```

Comment:
yeah, I don't really like for comprehensions for some reason... I'll change it back.